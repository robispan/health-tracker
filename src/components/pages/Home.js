import React from 'react';

import Card from '../assets/Card';
import StravaImg from '../assets/images/strava.jpg';
import NutriticsImg from '../assets/images/nutritics.png';
import RunkeeperImg from '../assets/images/runkeeper.jpg';
import TomtomImg from '../assets/images/tomtom.png';
import LifelogImg from '../assets/images/lifelog.png';

const Home = () => {
  return (
    <div>
      <h2 className='center upper'>Connect your apps</h2>

      <div className='card-container'>
        <Card imageUrl={StravaImg} />
        <Card imageUrl={NutriticsImg} />
        <Card imageUrl={RunkeeperImg} />
        <Card imageUrl={TomtomImg} />
        <Card imageUrl={LifelogImg} />
      </div>
    </div>
  );
};

export default Home;
