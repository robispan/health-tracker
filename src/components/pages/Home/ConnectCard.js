import React from 'react';

const Card = ({ imageUrl, connected, onConnect }) => {
  const classList = ['connect-card-link'];
  if (connected) classList.push('connected');

  return (
    <div className='connect-card'>
      <div
        className='connect-card-image'
        style={{ backgroundImage: `url(${imageUrl})` }}
      ></div>
      <a
        href='#'
        onClick={connected ? null : onConnect}
        className={classList.join(' ')}
      >
        {connected ? 'Connected' : 'Connect'}
      </a>
    </div>
  );
};

export default Card;
