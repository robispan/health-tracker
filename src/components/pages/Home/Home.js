import React from 'react';

import ConnectCard from './ConnectCard';
import StravaImg from '../../assets/images/strava.jpg';
import NutriticsImg from '../../assets/images/nutritics.png';
import RunkeeperImg from '../../assets/images/runkeeper.jpg';
import TomtomImg from '../../assets/images/tomtom.png';
import LifelogImg from '../../assets/images/lifelog.png';

const Home = () => {
  return (
    <div>
      <h2 className='center upper'>Connect your apps</h2>

      <div className='connect-card-grid'>
        <ConnectCard imageUrl={StravaImg} />
        <ConnectCard imageUrl={NutriticsImg} />
        <ConnectCard imageUrl={RunkeeperImg} />
        <ConnectCard imageUrl={TomtomImg} />
        <ConnectCard imageUrl={LifelogImg} />
      </div>
    </div>
  );
};

export default Home;
