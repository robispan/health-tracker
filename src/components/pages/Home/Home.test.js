import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import StravaContext from '../../../context/strava/stravaContext';
import LoadingContext from '../../../context/loading/loadingContext';
import RunKeeperContext from '../../../context/runKeeper/runKeeperContext';

import Home from './Home';

afterEach(cleanup);

test('renders correctly without context data', () => {
	const { asFragment } = render(
		<StravaContext.Provider value={{ stravaData: null }}>
			<LoadingContext.Provider value={{ loading: false }}>
				<RunKeeperContext.Provider value={{ getHealthData: null }}>
					<Home />
				</RunKeeperContext.Provider>
			</LoadingContext.Provider>
		</StravaContext.Provider>
	);
	expect(asFragment()).toMatchSnapshot();
});

test('renders correctly with context data', () => {
	const { asFragment } = render(
		<StravaContext.Provider value={{}}>
			<LoadingContext.Provider value={{}}>
				<RunKeeperContext.Provider value={{}}>
					<Home />
				</RunKeeperContext.Provider>
			</LoadingContext.Provider>
		</StravaContext.Provider>
	);
	expect(asFragment()).toMatchSnapshot();
});
