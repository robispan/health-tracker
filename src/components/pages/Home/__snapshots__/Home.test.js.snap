// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with context data 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="center upper"
    >
      Connect your apps
    </h2>
    <div
      class="grid"
    >
      <div
        class="connect-card"
      >
        <div
          class="connect-card-image"
          style="background-image: url(strava.jpg);"
        />
        <a
          class="connect-card-button connected"
          href="#"
        >
          Disconnect
        </a>
      </div>
      <div
        class="connect-card"
      >
        <div
          class="connect-card-image"
          style="background-image: url(nutritics.png);"
        />
        <a
          class="connect-card-button"
          href="#"
        >
          Connect
        </a>
      </div>
      <div
        class="connect-card"
      >
        <div
          class="connect-card-image"
          style="background-image: url(runkeeper.jpg);"
        />
        <a
          class="connect-card-button"
          href="#"
        >
          Connect
        </a>
      </div>
      <div
        class="connect-card"
      >
        <div
          class="connect-card-image"
          style="background-image: url(tomtom.png);"
        />
        <a
          class="connect-card-button"
          href="#"
        >
          Connect
        </a>
      </div>
      <div
        class="connect-card"
      >
        <div
          class="connect-card-image"
          style="background-image: url(lifelog.png);"
        />
        <a
          class="connect-card-button"
          href="#"
        >
          Connect
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders correctly without context data 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="center upper"
    >
      Connect your apps
    </h2>
    <div
      class="grid"
    >
      <div
        class="connect-card"
      >
        <div
          class="connect-card-image"
          style="background-image: url(strava.jpg);"
        />
        <a
          class="connect-card-button"
          href="#"
        >
          Connect
        </a>
      </div>
      <div
        class="connect-card"
      >
        <div
          class="connect-card-image"
          style="background-image: url(nutritics.png);"
        />
        <a
          class="connect-card-button"
          href="#"
        >
          Connect
        </a>
      </div>
      <div
        class="connect-card"
      >
        <div
          class="connect-card-image"
          style="background-image: url(runkeeper.jpg);"
        />
        <a
          class="connect-card-button"
          href="#"
        >
          Connect
        </a>
      </div>
      <div
        class="connect-card"
      >
        <div
          class="connect-card-image"
          style="background-image: url(tomtom.png);"
        />
        <a
          class="connect-card-button"
          href="#"
        >
          Connect
        </a>
      </div>
      <div
        class="connect-card"
      >
        <div
          class="connect-card-image"
          style="background-image: url(lifelog.png);"
        />
        <a
          class="connect-card-button"
          href="#"
        >
          Connect
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
